clients/src/main/scala/io/treeverse/clients/VarInt.scala

```diff
@@ -0,0 +1,38 @@
+package io.treeverse.clients
+
+import java.io.DataInput
+import java.io.IOException
+
+
+/**
+ * from https://github.com/addthis/stream-lib/blob/master/src/main/java/com/clearspring/analytics/util/Varint.java
+ */
+object VarInt {
+  @throws[IOException]
+  def readSignedVarLong(in: DataInput): Long = {
+    val raw = readUnsignedVarLong(in)
+    // This undoes the trick in writeSignedVarLong()
+    val temp = (((raw << 63) >> 63) ^ raw) >> 1
```

Comment:
love it.

Reply:
A classic.  Only needed in JVMs and similar "unsigned is too difficult".  Zigzag usually looks something like this (e.g. in Go):
```go
	ux := uint64(x) << 1
	if x < 0 {
		ux = ^ux
	}
```
which has the advantage of not having to pick between `>>` and `>>>` while remaining branch-free...